src/components/selector_menu/index.js

```diff
@@ -128,23 +126,30 @@ var SelectorMenu = React.createClass({
       return;
     }
 
-    var visible = _.map(fuzzy.filter(filterBy, this.state.visible), function(result) {
-      return result.string;
-    });
+    var visible = _.pluck(fuzzy.filter(filterBy, this.getOptionNames()), 'string');
 
     this.setState({
       visible: visible
     });
   },
 
+  componentWillReceiveProps: function(nextProps) {
+    if (nextProps.optionsList.length !== this.props.optionsList.length) {
```

Comment:
In python, I'd do something like:

`{x.title for x in props} + {x.name for x in props}` which would give me a set of all unique names or titles in the props. Then I could compare this set with another known set of the other props. If it only had one of title/name pairs, the None would still work out.